azure/durable_functions/entity.py

```diff
@@ -0,0 +1,114 @@
+from .models import DurableEntityContext
+from .models.entities import OperationResult, EntityState
+from datetime import datetime
+from typing import Callable, Any, List, Dict
+
+
+class Entity:
+    """Durable Entity Class.
+
+    Responsible for execuitng the user-defined entity function.
+    """
+
+    def __init__(self, entity_func: Callable[[DurableEntityContext], None]):
+        """Create a new entity for the user-defined entity.
+
+        Responsible for executing the user-defined entity function
+
+        Parameters
+        ----------
+        entity_func: Callable[[DurableEntityContext], Generator[Any, Any, Any]]
+            The user defined entity function
+        """
+        self.fn: Callable[[DurableEntityContext], None] = entity_func
+
+    def handle(self, context: DurableEntityContext, batch: List[Dict[str, Any]]) -> str:
+        """Handle the execution of the user-defined entity function.
+
+        Loops over the batch, which serves to specify inputs to the entity,
+        and collects results and generates a final state, which are returned.
+
+        Parameters
+        ----------
+        context: DurableEntityContext
+            The entity context of the entity, which the user interacts with as their Durable API
+
+        Returns
+        -------
+        str
+            A JSON-formatted string representing the output state, results, and exceptions for the
+            entity execution.
+        """
+        response = EntityState(results=[], signals=[])
+        for packet in batch:
+            result: Any = None
+            is_error: bool = False
+            start_time: datetime = datetime.now()
+
+            try:
+                # populate context
+                operation = packet["name"]
+                if operation is None:
+                    raise Exception("Entity operation is unassigned.")
```

Comment:
Considering this exception should never happen, as it would require us to change the schema, we should make this exception method as useful to helping us diagnose the schema break as possible. Something like "Entity operation was missing a name field".

Reply:
Also, this exception is caught in the catch block below, which I don't believe we want...

Reply:
I agree, and have incorporated, your suggestion about the Exception message.

As for catching the exception in the block below, I actually believe this is what we want. This allows us to report the exception back to the orchestrator.

Reply:
I'm not sure reporting this back as an `OperationResult` error is the best choice here. `is_error` being returned as a part of the schema to me indicates a user code failure, not a fundamental breakdown between the extension and this SDK schema like this would suggest.

I would almost rather have some catastrophic uncaught exception than have the extension gracefully handle this.

Reply:
Hmm, I think there's room for debate here but I'll default to your suggestion and make a note to discuss this further at a later date.

I'll introduce an internal exception type for us to re-throw if detected